### PR TITLE
Use aws-iam-authenticator when it's installed

### LINKED ISSUE #147: detect which authenticator command to use in kubeconfig
We've merged #117, which recognises new name of the authenticator command, but we still write kubecofig file for `heptio-authenticator-aws`. Ideally, we should keep the default behaviour as is, mostly because there are still no officially released binaries (https://github.com/kubernetes-sigs/aws-iam-authenticator/issues/119).

Simplest way to do this would be by calling `exec.LookupPath("aws-iam-authenticator")` from `WithExecHeptioAuthenticator` (which we should probably rename). We shouldn't have to do anything more sophisticated, as we won't need to supporting both authenticator versions for too long.

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

Fixes #147.

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [ ] Added tests that cover your change (if possible)
- [x] All tests passing (i.e. `make tests`)
- [x] Added/modified documentation as required (such as the README)
- [ ] Added yourself to the humans.txt file

### CODE REVIEW

**Comment by errordeveloper** [comment]
@richardcase once we merge this, let's tag 0.1.1 🚢!

File: pkg/eks/auth.go [hunk 1]
```diff
@@ -186,13 +187,13 @@ func (c *ClusterProvider) NewClientConfig() (*ClientConfig, error) {
...
 
-func (c *ClientConfig) WithExecHeptioAuthenticator() *ClientConfig {
+func (c *ClientConfig) WithExecAuthenticator() *ClientConfig {
 	clientConfigCopy := *c
 
 	x := clientConfigCopy.Client.AuthInfos[c.ContextName]
 	x.Exec = &clientcmdapi.ExecConfig{
 		APIVersion: "client.authentication.k8s.io/v1alpha1",
-		Command:    "heptio-authenticator-aws",
+		Command:    utils.DetectAuthenticator(),
```
Comment (richardcase) [inline pkg/eks/auth.go]: Nice, its great kops has this already.

File: pkg/utils/kubectl.go [hunk 2]
```diff
@@ -89,16 +93,39 @@ func CheckAllCommands(kubeconfigPath string, isContextSet bool, contextName stri
...
+		path, err := exec.LookPath(cmd)
 		if err == nil {
-			// binary was found
-			logger.Debug("%s: %q", bin, path)
+			// command was found
+			logger.Debug("%s: %q", cmd, path)
 			return nil
 		}
 	}
 	return fmt.Errorf("neither aws-iam-authenticator nor heptio-authenticator-aws are installed")
```
Comment (richardcase) [inline pkg/utils/kubectl.go]: Feel free to ignore this one! Would be good if the error message used the constants defined so there is less chance of deviation of error message from the commands.

Comment (errordeveloper) [reply pkg/utils/kubectl.go]: Yes, I thought about this, but we will soon care less about the legacy command, so I decided not too worry.

File: pkg/eks/auth.go (hunk 1 above)
Comment (errordeveloper) [reply pkg/eks/auth.go]: This is in our utils package, which now wraps the single kops utils function we use.

**Pull request merged**